android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -101,7 +117,38 @@ protected void onResume() {
         realmRepo.getSessionsByLocation(location)
                 .addChangeListener((sessions, orderedCollectionChangeSet) -> {
                     mSessions.clear();
+                    String upcomingTitle = "";
+                    String track = "";
+                    String color = null;
```

Comment:
If you declare it outside, it's scope leakage. And about the loop, it's already creating a new date object on each iteration